### PR TITLE
Site Settings: Move General settings to a separate section

### PR DESCRIPTION
**Pull request by tyxla**
This PR moves the General settings section to a separate chunk. This is the last large settings section that was not separated from the main settings chunk yet. Once we land this, we can perform several more cleanups within the `site-settings` code that are recommended after the recent changes.

Currently in production we have one `settings` chunk that is over 600K:

This PR splits out the general section, so the resulting chunks are now close to half that size each:

In more detail, the PR does the following:

* Moves general section to a separate chunk, `settings-general`.
* Cleans up everything unnecessary from the `settings` section.
* Preloads the `settings-general` section on hovering the General item of the settings navigation. With this change, all of the settings navigation items are now preloaded on hover.
* Preloads the `settings-general` section on hovering the Settings item of the sidebar navigation (previously, the `settings` section was preloaded).
* Suggests a WP logo placeholder while sites are being loaded (previously we had a blank screen)

To test:

* Checkout this branch.
* Go to http://calypso.localhost:3000/settings/ and verify it redirects to /settings/general and asks you to pick a site. Pick a site and verify it loads the general section correctly.
* Test the following paths with both a JP and WP.com site and verify there are no regressions with them:
  * http://calypso.localhost:3000/settings/general/$site
  * http://calypso.localhost:3000/settings/delete-site/$site
  * http://calypso.localhost:3000/settings/start-over/$site
  * http://calypso.localhost:3000/settings/theme-setup/$site
* Play with settings in the General section and verify they are stored and retrieved correctly.
* Go to Writing settings and refresh the page. Verify the `settings-general` chunk is preloaded correctly when hovering the General settings nav menu item for the first time.
* Go to Posts and refresh the page. Verify the `settings-general` chunk is preloaded correctly when hovering the Settings settings sidebar menu item for the first time.
* Verify the WP logo placeholder appears properly in the General tab if the sites are not loaded yet.

### CODE REVIEW

**Comment by oskosk** [comment]
Awesomeness!!!

File: client/my-sites/sidebar/sidebar.jsx [hunk 1]
```diff
@@ -447,7 +447,7 @@ export class MySitesSidebar extends Component {
 				link={ siteSettingsLink }
 				onNavigate={ this.onNavigate }
 				icon="cog"
-				preloadSectionName="settings"
+				preloadSectionName="settings-general"
```
Comment (seear) [inline client/my-sites/sidebar/sidebar.jsx]: Did a quick test—I see `settings-general.js` load when I hover, then `settings.js` loads when I click. So we could have a slight regression here unless we can preload two sections.

File: client/my-sites/site-settings/main.jsx [hunk 2]
```diff
@@ -28,15 +27,13 @@ export class SiteSettingsComponent extends Component {
 	};
 
 	static defaultProps = {
-		section: 'general'
+		section: 'guidedTransfer'
```
Comment (seear) [inline client/my-sites/site-settings/main.jsx]: This default prop is a red flag. Is it the case that this component needs to change to become `<GuidedTransfer/>`?

Comment (tyxla) [inline client/my-sites/site-settings/main.jsx]: Yeah, this was on my immediate list after I land this PR. I didn't want to bloat it too much, though. If you feel it's right, I can address it in this PR.

File: client/my-sites/sidebar/sidebar.jsx (hunk 1 above)
Comment (tyxla) [inline client/my-sites/sidebar/sidebar.jsx]: I don't think we need to preload `settings`, it'll load anyway every time when we load any of the settings sections anyway. Could you please describe the regression with more details?

File: client/my-sites/site-settings/main.jsx (hunk 2 above)
Comment (seear) [inline client/my-sites/site-settings/main.jsx]: Yep, a separate PR is best 👍

File: client/my-sites/sidebar/sidebar.jsx (hunk 1 above)
Comment (seear) [inline client/my-sites/sidebar/sidebar.jsx]: >  it'll load anyway every time when we load any of the settings sections anyway

This is the sidebar link, so `settings` may not yet be loaded. The performance regression is that on master, a hover over the sidebar link preloads everything necessary to display the general settings page. In this PR, we have split the section into two and are only preloading one of them.

**Comment by seear** [comment]
Some further discussion with @mtias, who is of the opinion that everything needed for `/general` should be in only one chunk, since it is the default landing page.

**Comment by tyxla** [comment]
It didn't take too much for you folks to convince me. It indeed makes sense for the general settings to be just one chunk, considering it's the default settings page. 

I'm closing this one, but will work on some related improvements and approaches to split other types of settings that make sense.

**Pull request closed without merging**